test_ffi/src/lib.rs

```diff
@@ -1,9 +1,60 @@
+use std::ffi::CStr;
+use std::os::raw::c_char;
+
 #[no_mangle]
 pub extern "C" fn print_something() {
   println!("something");
 }
 
 #[no_mangle]
-pub extern "C" fn add(a: u32, b: u32) -> u32 {
+pub unsafe extern "C" fn print_string(ptr: *const c_char) {
+  let cstr = CStr::from_ptr(ptr);
+  let name = cstr.to_str().unwrap();
+  println!("{}", name);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn print_buffer(ptr: *const u8, len: usize) {
```

Comment:
I've probably overlooked it, but wouldn't it be necessary with a bound check on the buffer to prevent someone from passing a invalid length for the raw pointer?